docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -11,410 +11,616 @@ The `ERC20.cairo` contract implements an approximation of https://eips.ethereum.
 * <<extensibility,Extensibility>>
 * <<presets,Presets>>
  ** <<erc20_basic,ERC20 (basic)>>
- ** <<erc20mintable,ERC20Mintable>>
- ** <<erc20pausable,ERC20Pausable>>
- ** <<erc20upgradeable,ERC20Upgradeable>>
 * <<api_specification,API Specification>>
- ** <<methods,Methods>>
+ ** <<erc20impl,ERC20Impl>>
   *** <<name,`name`>>
   *** <<symbol,`symbol`>>
   *** <<decimals,`decimals`>>
-  *** <<totalsupply,`totalSupply`>>
-  *** <<balanceof,`balanceOf`>>
+  *** <<total_supply,`total_supply`>>
+  *** <<balance_of,`balance_of`>>
   *** <<allowance,`allowance`>>
   *** <<transfer,`transfer`>>
-  *** <<transferfrom,`transferFrom`>>
+  *** <<transfer_from,`transfer_from`>>
   *** <<approve,`approve`>>
  ** <<events,Events>>
-  *** <<transfer_event,`Transfer (event)`>>
-  *** <<approval_event,`Approval (event)`>>
+  *** <<Transfer,`Transfer`>>
+  *** <<Approval,`Approval`>>
+ ** <<non-standard-external,Non-standard external>>
+  *** <<increase_allowance,`increase_allowance`>>
+  *** <<decrease_allowance,`decrease_allowance`>>
+ ** <<internalimpl,InternalImpl>>
+  *** <<initializer,initializer>>
+  *** <<internal_increase_allowance,_increase_allowance>>
+  *** <<internal_decrease_allowance,_decrease_allowance>>
+  *** <<internal_mint,_mint>>
+  *** <<internal_burn,_burn>>
+  *** <<internal_approve,_approve>>
+  *** <<internal_transfer,_transfer>>
+  *** <<internal_spend_allowance,_spend_allowance>>
 
 == Interface
```

Comment:
I think the rest of the methods (like `increase_allowance`) should appear here somehow as well. I don't think we should document the ERC20 interface, but our implementation interface (maybe specifying what methods are the ERC20 standard and what are not, but all of them should appear documented IMO).

Reply:
> I think the rest of the methods (like `increase_allowance`) should appear here somehow as well.

I put the non standard methods in the `Non-standard external` section (just under `Events`)—unless I'm misunderstanding your comment

> I don't think we should document the ERC20 interface, but our implementation interface (maybe specifying what methods are the ERC20 standard and what are not, but all of them should appear documented IMO)

I had a similar thought with just documenting our implementation and insert the `ERC20ABI` as the introductory interface. I'm open to the change if you really want to push for it. The OZ-contracts docs includes the [IERC20](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20) and the [ERC20](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#ERC20) implementation interface. I suppose we could include something like `IERC20 standard method that...` in the appropriate method descriptions

Reply:
as in the solidity-docs, `IERC20` and `ERC20ABI` are properly documented in the API section. i agree with Eric's comment on the main section interface. We should document our module (today it's a preset used a module, tomorrow it'll be a component), and mention it implements `IERC20` and exposes `ERC20ABI`.